### PR TITLE
Feature/oauth registration

### PR DESCRIPTION
**Pull request by connor4312**
This implements OAuth registration into the developers site. It requires config changes. I recursively uses oauth to authorize with the main site, so the client record to the developers site will have to be inserted manually.

It works, but could use some improvements and such over time. Also our setup with React is really painful to develop with >.<

### CODE REVIEW

File: src/components/oauth_manage.jsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    );
+  }
+}
+
+class Help extends React.Component {
+  render() {
+    return (
+      <div>
+        <h2>Get Started with OAuth</h2>
+        <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Duis condimentum metus eu lacus laoreet elementum. Morbi vehicula luctus tortor, eget gravida felis condimentum non. Morbi lobortis rhoncus tellus in ultrices. Duis id neque vitae diam elementum commodo et eget risus. Mauris vitae scelerisque urna. Ut in porttitor neque. Nunc euismod augue id velit molestie tristique. Curabitur rhoncus vestibulum quam.</p>
```
Comment (ttaylorr) [inline src/components/oauth_manage.jsx]: ?

File: src/components/routes.jsx [hunk 2]
```diff
@@ -8,6 +8,7 @@ import APIReferencePage from "./api_reference_page.js";
 import GuidePage from "./guide_page.js";
 import WidgetPage from "./widget_page_component.js";
 import ChatProtoPage from "./chatproto_page_component.js";
+const OAuthManage = require('./oauth_manage');
```
Comment (ttaylorr) [inline src/components/routes.jsx]: Can this be an `import`?

**Comment by ttaylorr** [comment]
:+1:

File: src/components/oauth_manage.jsx (hunk 1 above)
Comment (connor4312) [inline src/components/oauth_manage.jsx]: Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum sed nibh euismod, luctus eros in, dictum sem. Duis dapibus sodales urna, eget maximus ligula aliquam et. Suspendisse risus diam, scelerisque at scelerisque laoreet, ullamcorper consequat lectus. Aliquam maximus ante id arcu cursus, ac placerat metus suscipit. Cras ornare sed sem sit amet malesuada. Nullam arcu lorem, ullamcorper eu facilisis eget, bibendum vestibulum eros. Aliquam imperdiet id justo in tincidunt. Integer vestibulum risus felis, ac eleifend justo commodo ut. Nam vestibulum libero id gravida consequat. Ut tellus libero, tincidunt ut leo malesuada, sollicitudin convallis mi. Ut condimentum est nec lacus pharetra, in imperdiet nisl egestas. Nam ut lectus erat. Morbi rutrum ex in dolor rhoncus, at ultricies risus mattis.

File: src/components/routes.jsx (hunk 2 above)
Comment (connor4312) [inline src/components/routes.jsx]: Derp, yea, don't know what/if I was thinking

**Comment by JamyDev** [comment]
Looks good appart from the jsx stuff, I'm not familiar with the syntax :)

**Comment by MCProhosting-Matt** [comment]
Just waiting on the change to imports and then we're good to go.

**Pull request merged**